MoviesExample/app/src/main/java/com/novoda/demo/movies/MoviesViewModel.java

```diff
@@ -0,0 +1,51 @@
+package com.novoda.demo.movies;
+
+import android.arch.core.util.Function;
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.Transformations;
+import android.arch.lifecycle.ViewModel;
+
+import com.novoda.demo.movies.model.Movie;
+import com.novoda.demo.movies.model.Video;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MoviesViewModel extends ViewModel {
+
+    private final MovieService movieService;
+    private MoviesSate moviesSate = new MoviesSate(new ArrayList<Movie>(), 1);
+
+    MoviesViewModel(MovieService movieService) {
+        this.movieService = movieService;
+    }
+
+    public LiveData<MoviesSate> moviesLiveData() {
+        return Transformations.map(movieService.loadMore(moviesSate.pageNumber()),
+                new Function<List<Movie>, MoviesSate>() {
+                    @Override
+                    public MoviesSate apply(List<Movie> input) {
+                        List<Movie> movies = moviesSate.movies();
+                        movies.addAll(input);
+                        moviesSate = new MoviesSate(movies, moviesSate.pageNumber() + 1);
+                        return moviesSate;
```

Comment:
This API to domain mapping code is the same we had in the old `MovieService` class: https://github.com/novoda/android-demos/pull/41/files#diff-5643575e943cc4ee1d7bbb5b78d39395L49